app/javascript/pages/Settings/Payments/Show.tsx

```diff
@@ -276,8 +173,36 @@ const PaymentsPage = (props: Props) => {
   const [showNewBankAccount, setShowNewBankAccount] = React.useState(!props.bank_account_details.account_number_visual);
 
   React.useEffect(() => {
-    if (errorMessage) formRef.current?.scrollIntoView({ behavior: "smooth" });
-  }, [errorMessage]);
+    if ((errors?.base && errors.base.length > 0) || clientErrorMessage) {
+      formRef.current?.scrollIntoView({ behavior: "smooth" });
+    }
+  }, [errors, clientErrorMessage]);
+
+  React.useEffect(() => {
+    setComplianceInfo(props.compliance_info);
+  }, [props.compliance_info]);
+
+  React.useEffect(() => {
+    setBankAccount(props.bank_account_details.bank_account);
+    setShowNewBankAccount(!props.bank_account_details.account_number_visual);
+  }, [props.bank_account_details]);
+
+  React.useEffect(() => {
+    setPaypalEmailAddress(props.paypal_address);
+  }, [props.paypal_address]);
+
+  React.useEffect(() => {
+    const newPayoutMethod: PayoutMethod = props.stripe_connect.has_connected_stripe
+      ? "stripe"
+      : props.bank_account_details.show_bank_account && props.bank_account_details.is_a_card
+        ? "card"
+        : props.bank_account_details.account_number_visual !== null
+          ? "bank"
+          : props.bank_account_details.show_paypal
+            ? "paypal"
+            : "bank";
+    setSelectedPayoutMethod(newPayoutMethod);
+  }, [props.stripe_connect.has_connected_stripe, props.bank_account_details]);
```

Comment:
Added these useEffect hooks to sync local component state with props that may change after form submissions.

React `useState` only uses props as initial values and doesn't automatically update when props change. After a form submission (e.g., country update), Inertia re-renders with new props from the server, but the local state remains stale.

These useEffect hooks watch for prop changes and sync the local state accordingly. This ensures that when server-side updates occur (like country changes that cascade to states, compliance info, bank account details, etc.), the UI reflects the new values immediately.

This is especially critical for country updates where multiple dependent fields change simultaneously and users need to see the updated form fields right away.